CHANGELOG.md

```diff
@@ -10,7 +10,8 @@
 ### API changes
 
 - Removed `SentryOptionsExtensions` class - all the public methods moved directly to `SentryOptions` ([#3195](https://github.com/getsentry/sentry-dotnet/pull/3195))
-
+- Sentry now uses System.Diagnostics.DiagnosticSource internally for tracing. Any traces created by Sentry can now be captured via OpenTelemetry. If you are using .NET Core 3.1 or earlier, you will need to add the `Sentry.DiagnosticSource` package to your solution to enable tracing. ([#3238](https://github.com/getsentry/sentry-dotnet/pull/3238))
```

Comment:
The only way to avoid the new dependency for netstandard and netfx would be to continue to use our existing tracing mechanism for those targets for tracing in all of our integrations. 

This would require quite a bit of extra work to build and maintain... but I guess we'd be making the new OTel behind the scenes instrumentation opt-in only for netfx and netstandard at that point. The only reason I can think why they might do this is so that they could capture Sentry spans from OTel (and show them in something like their Aspire dashboard).

Reply:
> The only way to avoid the new dependency for netstandard and netfx would be to continue to use our existing tracing mechanism for those targets for tracing in all of our integrations.

We don't need to add anything to netstandard at this point, it's just here to have some extra time to phase out. 

But .NET Framework is here to stay. So we need to consider that 

> This would require quite a bit of extra work to build and maintain... but I guess we'd be making the new OTel behind the scenes instrumentation opt-in only for netfx and netstandard at that point.

Yeah makes sense to me.

Seems like it's OK to keep support only to .NET (6+) then. And .NET Framework continues as-is?

Reply:
k, I've created two concrete implementations of the Tracing interfaces:
1. One using the `System.Diagnostics.DiagnosticSource` classes (`ActivitySource`, `Activity` etc.)
2. Another that uses Sentry's tracing classes

The `Activity` classes will be used by default for `net5.0` targets and later... otherwise they're opt in (via the `Sentry.DiagnosticsSource` package). 

So no additional dependencies for anyone, unless they explicitly opt in via `Sentry.DiagnosticsSource`